src/FakeItEasy/Core/FakeManager.cs

```diff
@@ -309,6 +322,35 @@ private void UpdateLastSequenceNumber(int sequenceNumber)
                    sequenceNumber != Interlocked.CompareExchange(ref this.lastSequenceNumber, sequenceNumber, last));
         }
 
+        private void ReplaceUserRules(IEnumerable<CallRuleMetadata>? rules)
+        {
+            // ReSharper disable once InconsistentlySynchronizedField
+            this.allUserRules = rules is null
+                ? new LinkedList<CallRuleMetadata>()
+                : new LinkedList<CallRuleMetadata>(rules);
+        }
+
+        private void ReplaceRecordedCalls(CompletedFakeObjectCall[]? calls)
+        {
+            var newRecordedCalls = calls is null
+                ? new ConcurrentQueue<CompletedFakeObjectCall>()
+                : new ConcurrentQueue<CompletedFakeObjectCall>(calls);
+            this.recordedCalls = newRecordedCalls;
+
+            var lastCall = calls?.LastOrDefault();
+            if (lastCall is not null && lastCall.SequenceNumber > this.lastSequenceNumber)
+            {
+                this.UpdateLastSequenceNumber(lastCall.SequenceNumber);
+            }

```

Comment:
This would not have occurred to me to do. As it is, I'm not sure it's needed. How would ever a sequence number from a recorded call be bigger than our current sequence number?